### PR TITLE
add a RedirectBlock

### LINKED ISSUE #3303: Add a Redirect Block
**Describe the bug**

If we build a content structure like this:

/Plone/communications-department
/Plone/communications-department/news
/Plone/communications-department/events
/Plone/communications-department/press-releases

In Plone we can add a "Link" item, set it as a "default page" for `communications-department` folder and get a redirect to, say, `/Plone/communications-department/news`, this way we can group several folders under a folder and get the users redirected.

In plain Volto this is not possible.

A "Redirect Block" could help here. If we add a "Redirect Block" to the `communications-department` then it will work like in Classic Plone.

### PR DESCRIPTION
**Pull request by erral**
Fixes #3303 

If a RedirectBlock is added to any content, the page issues a 302 Redirect for anonymous users.

### CODE REVIEW

**Review by nileshgulia1 — COMMENTED** [review]
Nice. It would be great to have some tests for this new block components.

File: src/components/manage/Blocks/Redirect/schema.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        title: 'Default',
+        fields: ['remoteUrl'],
+      },
+    ],
+    properties: {
+      remoteUrl: {
+        title: intl.formatMessage(messages.remoteUrl),
+        widget: 'object_browser',
+        mode: 'link',
+        llowExternals: true,
```
Comment (sneridagh) [inline src/components/manage/Blocks/Redirect/schema.js:31]: typo!

**Review by sneridagh — CHANGES_REQUESTED** [review]
I'm ok with this one, please fix the typo!

**Comment by fredvd** [comment]
As mentionned in the Volto team meeting: creating 'functional' blocks that are not for the blocks layout visual but modify the content item behavior sounds more like a behavior or in this case sticking an instance behavior to the context.   So then you have functional and visual blocks. 

Doing this seems like starting a visual programming editor/tool where you drag 'actions' onto the canvas and then they are active. I'm afraid that mixing these different purposes for the blocks editor will confuse editors with a too big conceptual load of how to understand the UI. 

Understanding the use case: is this a 'default page' light returning to Volto, so you can stick it on a folderish page?

**Comment by erral** [comment]
Sorry I was not able to attend the Volto team meeting to discuss this.

This PR tries to bring to Volto the following feature that exists in Classic Plone: 

- Create a Folder
- Create a link inside a folder
- Select the link as a default page for the Folder

With this scenario one can add content in the site, place it on menus, or whatever, and get redirects to another place of the site or even to external site. The main use case is the one described in the issue.

**Comment by erral** [comment]
I added some tests

**Pull request closed without merging**